llama_stack/providers/utils/tools/ttl_dict.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+import time
+from threading import RLock
+from typing import Any
+
+
+class TTLDict(dict):
```

Comment:
Do we want to maintain our own `TTLDict` implementation? Or would we prefer to pull in another dependency, such as `cachetools`, and to use its [TTLCache](https://cachetools.readthedocs.io/en/latest/#cachetools.TTLCache)?

I only ask because as a reviewer, I cannot immediately tell whether this TTLDict implementation is correct, thread-safe, performant, etc.

Reply:
For `cachetools` the implementations are explicitly not thread safe:
> Note:  Please be aware that all these classes are not thread-safe. Access to a shared cache from multiple threads must be properly synchronized, e.g. by using one of the memoizing decorators with a suitable lock object. 

(source: https://cachetools.readthedocs.io/en/latest/#cache-implementations)

I don't mind looking for another dependency that is cache safe, let me know if you would prefer that to having our own implementation here

Reply:
I don't think we need a dependency for something so simple. My first inclination would be to not even have a TTL dict class (until a _second_ usecase arrives) and just inline whatever minimal functionality is needed in the calling code. But this is definitely nicer and also fine.